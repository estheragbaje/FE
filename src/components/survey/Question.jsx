import React from 'react';
import PropTypes from 'prop-types';
import { StyledQuestion, Close } from './SurveyStyles';
import OptionSelect from '../common/OptionSelect';
import TextArea from '../common/TextArea';

const typeOptions = [
  {
    name: 'Text',
    value: 'text',
  },
  {
    name: 'Rating',
    value: 'rating',
  },
];

const Question = ({
  question,
  type,
  index,
  handleChangeQuestion,
  removeQuestion,
}) => (
  <StyledQuestion className="z-depth-1">
    <Close
      type="button"
      className="remove-question"
      id="closeQuestion"
      onClick={() => removeQuestion(index)}
    >
      âœ•
    </Close>

    <TextArea
      label="Question"
      name="question"
      value={question}
      onChange={(e) => handleChangeQuestion(index, e)}
      className="form-control mb-4"
      id="question"
      rows="3"
    />

    <OptionSelect
      label="Question type"
      name="type"
      value={type}
      options={typeOptions}
      onChange={(e) => handleChangeQuestion(index, e)}
      className="browser-default custom-select mb-4"
      id="type"
    />
  </StyledQuestion>
);

Question.propTypes = {
  question: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  handleChangeQuestion: PropTypes.func.isRequired,
  removeQuestion: PropTypes.func.isRequired,
};

export default Question;
