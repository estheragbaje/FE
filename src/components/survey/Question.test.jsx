import React from 'react';
import { shallow, mount } from 'enzyme';

import Question from './Question';
import TextArea from '../common/TextArea';

describe('<Question />', () => {
  const props = {
    handleChangeQuestion: jest.fn(),
    removeQuestion: jest.fn(),
    question: jest.mock(),
    history: { push: jest.fn() },
  };
  it('renders the Question component correctly', () => {
    const wrapper = shallow(<Question />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the mock onSubmit function', () => {
    const wrapper = mount(<Question {...props} />);
    expect(wrapper.find(TextArea).length).toBe(1);
  });

  it('should call the mock handleChangeQuestion function', () => {
    const wrapper = mount(<Question {...props} />);

    wrapper
      .find('.form-control .mb-4')
      .simulate('change', { preventDefault() {} });
    expect(props.handleChangeQuestion).toBeCalled();
  });

  it('should call the mock handleChangeQuestion function', () => {
    const wrapper = mount(<Question {...props} />);

    wrapper.find('#type').simulate('change', { preventDefault() {} });
    expect(props.handleChangeQuestion).toBeCalled();
  });
});
