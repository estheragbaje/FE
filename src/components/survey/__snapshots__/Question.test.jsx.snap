// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Question /> renders the Question component correctly 1`] = `
<styled.div
  className="z-depth-1"
>
  <styled.button
    className="remove-question"
    id="closeQuestion"
    onClick={[Function]}
    type="button"
  >
    âœ•
  </styled.button>
  <TextArea
    className="form-control mb-4"
    id="question"
    label="Question"
    name="question"
    onChange={[Function]}
    rows="3"
  />
  <OptionSelect
    className="browser-default custom-select mb-4"
    id="type"
    label="Question type"
    name="type"
    onChange={[Function]}
    options={
      Array [
        Object {
          "name": "Text",
          "value": "text",
        },
        Object {
          "name": "Rating",
          "value": "rating",
        },
      ]
    }
  />
</styled.div>
`;
